example_client_test.go

```diff
@@ -36,28 +36,32 @@ func Example() {
 
 // Queue represents a connection to a specific queue.
 type Queue struct {
-	name          string
-	logger        *log.Logger
-	connection    *amqp.Connection
-	channel       *amqp.Channel
-	done          chan bool
-	notifyClose   chan *amqp.Error
-	notifyConfirm chan amqp.Confirmation
-	isConnected   bool
+	name            string
+	logger          *log.Logger
+	connection      *amqp.Connection
+	channel         *amqp.Channel
+	done            chan bool
+	notifyConnClose chan *amqp.Error
+	notifyChanClose chan *amqp.Error
+	notifyConfirm   chan amqp.Confirmation
+	isReady         bool
 }
 
 const (
 	// When reconnecting to the server after connection failure
 	reconnectDelay = 5 * time.Second
 
+	// When setting up the channel after a channel exception
+	resetupDelay = 2 * time.Second
```

Comment:
"resetup" is not really a word. I think "reinitialization" or "reinit" would be clearer and more familiar here.